mock-caas/src/main/java/com/sequenceiq/caas/grpc/service/MockGroupManagementService.java

```diff
@@ -16,120 +16,151 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 
-import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.DeleteWorkloadAdministrationGroupNameRequest;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.DeleteWorkloadAdministrationGroupNameResponse;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.GetWorkloadAdministrationGroupNameRequest;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.GetWorkloadAdministrationGroupNameResponse;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.GetWorkloadAdministrationGroupNameResponse.Builder;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.Group;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.ListGroupsRequest;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.ListGroupsResponse;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.SetWorkloadAdministrationGroupNameRequest;
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.SetWorkloadAdministrationGroupNameResponse;
 import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.Crn.ResourceType;
 import com.sequenceiq.cloudbreak.auth.altus.UmsRight;
 
 import io.grpc.stub.StreamObserver;
 
 @Service
 class MockGroupManagementService {
+
     private static final Logger LOGGER = LoggerFactory.getLogger(MockGroupManagementService.class);
 
     private static final String CM_ADMIN_RIGHT = "environments/adminClouderaManager";
 
+    private static final int NUM_USER_GROUPS = 5;
+
     @Inject
     private MockCrnService mockCrnService;
 
-    private final Map<String, Map<String, UserManagementProto.Group>> accountGroups = new ConcurrentHashMap<>();
+    private final Map<String, Map<String, Group>> accountWorkloadGroups = new ConcurrentHashMap<>();
```

Comment:
WAGs should be modeled as `WorkloadAdministrationGroup`s rather than `Group`s. WAGs don't have crns, only names, accountids, rights, and resources. Moving to `WorkloadAdministrationGroup` would probably make the WAG-related mock code more sensible.